### PR TITLE
PLAT-601-4 Implemented BigDecimalParser and Strings

### PR DESCRIPTION
**Pull request by oliver-richers**

### CODE REVIEW

File: platform-common-core/src/main/java/com/softicar/platform/common/core/number/parser/BigDecimalParser.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+			var integralPart = numberString.substring(0, decimalSeparatorIndex);
+			var fractionalPart = numberString.substring(decimalSeparatorIndex + decimalSeparator.length());
+			validateIntegralPart(integralPart);
+			validateFractionalPart(fractionalPart);
+		}
+	}
+
+	private void validateIntegralPart(String integralPart) {
+
+		String[] groups = integralPart.replace(locale.getDigitGroupSeparator(), ",").split(",");
```
Comment (thees-k) [inline platform-common-core/src/main/java/com/softicar/platform/common/core/number/parser/BigDecimalParser.java:99]: Why not just
`String[] groups = integralPart.split(locale.getDigitGroupSeparator());`
?

File: platform-common-core/src/main/java/com/softicar/platform/common/core/number/parser/BigDecimalParser.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+			validateIntegralPart(numberString);
+		} else {
+			var integralPart = numberString.substring(0, decimalSeparatorIndex);
+			var fractionalPart = numberString.substring(decimalSeparatorIndex + decimalSeparator.length());
+			validateIntegralPart(integralPart);
+			validateFractionalPart(fractionalPart);
+		}
+	}
+
+	private void validateIntegralPart(String integralPart) {
```
Comment (thees-k) [inline platform-common-core/src/main/java/com/softicar/platform/common/core/number/parser/BigDecimalParser.java:97]: The integral part may consist of digits and digitGroupSeparator only. No other characters are allowed. I think this should be validated.

File: platform-common-core/src/main/java/com/softicar/platform/common/core/number/parser/BigDecimalParser.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+					throw new SofticarUserException(CommonCoreI18n.DIGIT_GROUP_TOO_LONG);
+				}
+				if (i > 0 && groups[i].length() < DIGIT_GROUP_SIZE) {
+					throw new SofticarUserException(CommonCoreI18n.DIGIT_GROUP_TOO_SHORT);
+				}
+			}
+		}
+	}
+
+	private void validateFractionalPart(String fractionalPart) {
```
Comment (thees-k) [inline platform-common-core/src/main/java/com/softicar/platform/common/core/number/parser/BigDecimalParser.java:112]: The fractional part may consist of digits and decimalSeparator only. No other characters are allowed. I think this should be validated.

**Review by thees-k — CHANGES_REQUESTED** [review]
I think the methods `validateIntegralPart()` and `validateFractionalPart()` should be improved (please see the commented lines).

File: platform-common-core/src/main/java/com/softicar/platform/common/core/number/parser/BigDecimalParser.java (hunk 3 above)
Comment (oliver-richers) [reply platform-common-core/src/main/java/com/softicar/platform/common/core/number/parser/BigDecimalParser.java:112]: new later `new BigDecimal(numberString)` call will already catch that

**Review by thees-k — APPROVED** [review]

**Pull request merged**